### ISSUE TITLE
Approach for targeted DMS Map-Seq Strategy

### ISSUE THREAD

**Issue by nickforino** [dincarnato/RNAFramework#10]
Hi Danny, I'm on your support thread yet again. Hoping you are well.

I'm currently employing a targeted Map-Seq strategy for individual RNAs that I feel is suboptimal. My issue is arising at the rf-fold step.

My approach is to build custom reference sequences using bowtie2 for read alignment. I found that using the exact DNA sequence of my RNA as the reference would yield overall poor alignment results (0.1% alignment), however including some flanking intergenic DNA (~500 nt) on both sides of my target enabled near-complete alignment of my reads (95%). I suspect this is some behavior of how bowtie2 aligns reads. Counting and normalizing my data progresses without a problem, but the folding step discards my transcript because 'folding failed'.

I suspected folding was failing because including flanking sequence in my reference sequence produces hundreds of NaN values flanking my region of interest in the XML file. My next attempt was to use a FASTA of my exact RNA sequence in the rf-count step, in hopes of producing an XML without NaN values. This failed because the reference sequence in the input BAM header doesn't match the FASTA I provided.

My question to you is whether you feel my workflow can be optimized for the targeted Map-Seq strategy. If the optimal strategy involves aligning my reads to the exact sequence of my target RNA, what steps can I take to optimize the alignment?

Thanks so much for your time and advice Danny.
Best,
Nick

**Comment by dincarnato** [comment]
Hi Nick,

First of all, what does it mean that the folding "failed"? The program threw an exception, or you just did not get the wanted/expected result?

Second, I am going to make the educate guess that you are performing read alignment in the end-to-end mode, am I right? If that's the case, just go back to "real" reference you were using in the beginning, and enable the local alignment/soft clipping. You can rapidly test if that was the problem by using the --very-sensitive-local preset of bowtie2. To do so, just invoke rf-fold with the parameter -mp '--very-sensitive-local'.

Hope this helps!
Best,

Danny

**Comment by nickforino** [comment]
Hi Danny,

To your first point, sorry for not clarifying. Output of rf-fold returned this error message: 
`Folding RNA structures [Last: none]sh: line 1: 95207 Abort trap: 6`.
```
[+] Folding statistics:

  [*] Folded transcripts:    0
  [*] Discarded transcripts: 1 total
                             0 XML parsing failed
                             0 constraint file generation failed
                             0 FASTA file generation failed
                             1 folding failed
                             0 I/O error
```
To your second point about aligning using local alignment/soft clipping, this worked! Alignment rate was excellent, and I was able to generate a folding prediction using both Vienna RNA and RNAstructure.

Once again, I'm so very appreciative for your help Danny! Thanks for your continued support and taking the time to help me out.

Cheers,
Nick

**Comment by dincarnato** [comment]
Hi Nick,

great that mapping worked, and that this also solved your issue with the folding.
However, I would like to track down the error you got with rf-fold with the first prediction you made.
Can you please share the XML file you used and the exact command line so that I can reproduce the issue?

Thanks,

Danny

**Comment by nickforino** [comment]
Hi Danny,

I believe this issue stemmed from my RNAstructure version and isn't an RNAFramework bug. 
I didn't mention in my last message that in addition to trying the different alignment approach, I also updated my RNAstructure from the 5.6 beta version to the newest. I went back to the older XML file generated by aligning my reads to my target + flanking intergenic DNA, and was successfully able to execute rf-fold using the command that failed previously:

 `rf-fold -m 2 -p 4 -g TERC_region.xml`

I've attached the XML file for you to try, I suspect it will work for you.

Best,
Nick
[TERC_region.zip](https://github.com/dincarnato/RNAFramework/files/5914032/TERC_region.zip)
